source/configuration/modules/imfile.rst

```diff
@@ -69,9 +69,17 @@ disabled by default. It can explicitly be turned on or off via the
 State Files
 ...........
 Rsyslog must keep track of which parts of the monitored file
-are already processed. This is done in so-called "state files".
-These files are always created in the rsyslog working directory
-(configurable via $WorkDirectory).
+are already processed. This is done in so-called "state files" that
+are created in the rsyslog working directory. These files are read
+on startup to resume monitoring after a shutdown.
+
+**Note**: The ``PersistStateInterval`` parameter must be set, otherwise state
+files will NOT be created.
+
+The location of the rsyslog working directory is configurable via the
+``$WorkDirectory`` |FmtObsoleteName| format (limited to |FmtObsoleteName|
```

Comment:
I would completely drop the obsolete format. That only encourages use. Let's get rid of it everywhere except in the short cross-ref sections at the end of the config objects (e.g,. plugin like imuxsock).

Reply:
Fair enough. I was just attempting to balance the mention of the obsolete legacy format directive, thinking that we could loop back later as part of a focused effort to prune legacy directives.

However I think it makes good sense to replace use of the old directives with the new/advanced format equivalents as we find them.

Reply:
IMHO the obsolete statements inside doc text, tutorials etc only are there because nobody yet too the time to change them to advanced format. From the mailing list and issues I know that many people find it confusing to see this mix of obsolete and advanced statements (and I think they are right with that).

The cure actually is to change the statement and not mention the obsolete ones. I still think we should provide a very brief reference of old-to-new mapping, but not more than that. Whoever wants more info on the old statements should refer to old doc. One reason is that it often is really very hard to accurately describe what obsolete format does, because it is so counter-intuitive and context-depending (from a CS point of view, it's actually a context depending mes^h^h^hlanguage).

Reply:
> @rgerhards: IMHO the obsolete statements inside doc text, tutorials etc only are there because nobody yet took the time to change them to advanced format.

That and perhaps there was a question of whether the doc should include both or only the new format. I think we've clarified some of that with recent discussions, though it may be a good idea to create a separate issue that makes that clear and could serve as a good pointer if/when questions about the work comes up.

Reply:
> @rgerhards: The cure actually is to change the statement and not mention the obsolete ones. I still think we should provide a very brief reference of old-to-new mapping, but not more than that. 

Do you see that goal being accomplished by way of the "obsolete format directives" section at the bottom of module pages?